language/deepseek-r1/README.md

```diff
@@ -0,0 +1,140 @@
+# Mlperf Inference DeepSeek Reference Implementation
+
+## Docker
+
+The MLPerf DeepSeek reference implementation includes a comprehensive Docker launch system that supports multiple backends and provides advanced features like user management, persistent storage, and flexible configuration.
+
+### Launch Backend Specific Container
+
+Launch a Docker container with your preferred backend:
+
+```bash
+# Launch PyTorch backend
+./launch.sh --backend pytorch
+
+# Launch vLLM backend
+./launch.sh --backend vllm
+
+# Launch SGLang backend
+./launch.sh --backend sglang
+
+# See launch.sh for full list of args
+./launch.sh --backend vllm --gpu-count 2 --extra-mounts "/data:/data,/models:/models" --local-user 0
+```
+
+### Available Backends
+
+- **pytorch**: via [DeepSeek-Ai/DeepSeek-V3](https://github.com/deepseek-ai/DeepSeek-V3) (reference implementation by DeepSeek-Ai)
+- **vllm**: vLLM's LLM api-based inference
+- **sglang**: sglang's OpenAI endpoint-based inference
+
+## Backend-Specific Setup
+
+After launching any Docker container, run the setup script which automatically detects your backend:
+
+```bash
+# Automatic backend detection and setup
+setup.sh
+```
+
+The setup script creates a virtual environment and configures it differently based on the backend:
+
+#### All Backends
+- Virtual environment is **activated** after `setup.sh`
+- Activate backend-specific venv using `source .venv_[pytorch|vllm|sglang]/bin/activate` 
+- All commands are to be run using the virtual environment
+
+## Running Evaluations
+
+### PyTorch Backend (Distributed)
```

Comment:
We probably leave a note and saying this is going to take >7 days to finish on a single DGX system, and recommend the user to only use it for some sample validation not e2e runs